src/main/java/namecorp/camera_lock_on/compatibility/optionals/shoulderSurfing/ShoulderSurfingEntityPicker.java

```diff
@@ -0,0 +1,35 @@
+package namecorp.camera_lock_on.compatibility.optionals.shoulderSurfing;
+
+import org.jetbrains.annotations.Nullable;
+
+import namecorp.camera_lock_on.adapters.IEntityPickerAdapter;
+import namecorp.camera_lock_on.compatibility.optionals.AdditionalMods;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.hit.HitResult;
+
+public class ShoulderSurfingEntityPicker implements IEntityPickerAdapter {
+    private IEntityPickerAdapter vanillaEntityPickerAdapter;
+
+    public ShoulderSurfingEntityPicker(IEntityPickerAdapter vanillaEntityPickerAdapter) {
+        this.vanillaEntityPickerAdapter = vanillaEntityPickerAdapter;
+    }
+
+    @Override
+    public HitResult pick(double interactionRange, float tickDelta) {
+        ShoulderSurfingManager shoulderSurfingManager = AdditionalMods.shouderSurfing();
+        @Nullable HitResult result = shoulderSurfingManager.pick(interactionRange, tickDelta);
+        if (result == null) vanillaEntityPickerAdapter.pick(interactionRange, tickDelta);
+        return result;
+    }
+
+    @Override
+    public void LookAt(Entity entity, WorldRenderContext last) {
+        vanillaEntityPickerAdapter.LookAt(entity, last);
+        ClientPlayerEntity player = MinecraftClient.getInstance().player;
+        AdditionalMods.shouderSurfing().setCameraAngle(player.getYaw(), player.getPitch());
```

Comment:
@Exopandora could you give me some guidance? vanillaEntityPickerAdapter.LookAt calculate player's angles in a vanilla friendly way, but the result is not right compensated considering Shoulder Surfing offset (see demonstration video to more details).
Do you have any suggestions on how I can fix it?
Maybe I need to implement the angle calculations specifically for ShoulderSurfing